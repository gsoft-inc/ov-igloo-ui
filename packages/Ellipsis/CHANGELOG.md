# @igloo-ui/ellipsis

## 0.3.0

### Minor Changes

- 4846c59: Wrapped Igloo CSS in @layer to isolate it.

## 0.2.4

### Patch Changes

- cfb5619: Updated hopper token versions and button versions.

## 0.2.3

### Patch Changes

- 7d36815: Updated hopper token versions and igloo icon versions.

## 0.2.2

### Patch Changes

- a2df55d: Updated to latest Hopper Token dependency

## 0.2.1

### Patch Changes

- 9739155: Gave ColorPicker the new Wokleap look. Color now uses css variables to update its color. Fixed spacing issues in tooltip. Updated hopper token version.

## 0.2.0

### Minor Changes

- 36fbdcc: Rebrand of Ellipsis

## 0.1.11

### Patch Changes

- bf4ddef: Updated storybook version and linting. Made linting fixes in these components.

## 0.1.10

### Patch Changes

- a41e1d4: Fix: moved destructuring of props to function parameters so that storybook will automatically show the default value in the props table. It was done in Modal because it needed to send props to useDialog. In this case, @default annotation was used in comments.

## 0.1.9

### Patch Changes

- bf478b5: Added missing tests. Fixed missing code in stories when clicking 'Show code'. Updated Dropdown mock. Added new optional prop highlightToday to help with test snapshots.

## 0.1.8

### Patch Changes

- 6ee9c45: Updated anything causing errors or warnings in the browser console

## 0.1.7

### Patch Changes

- 0640409: udpate classnames to 2.3.2

## 0.1.6

### Patch Changes

- d971143: All tests were updated to use testing library and not enzyme

## 0.1.5

### Patch Changes

- 0a86f13: Update content of README

## 0.1.4

### Patch Changes

- 01fbda0: add posibility to forward a `ref`

## 0.1.3

### Patch Changes

- 8c8686e: Change @import of font.css file for a @use in SASS files

## 0.1.2

### Patch Changes

- afdaa6b: Fix a lib import error with capital cases

## 0.1.1

### Patch Changes

- 4a256e3: IE11 compatibility

## 0.1.0

### Minor Changes

- e90774d: ## Ellipsis component

  Add Ellipsis component to allow text truncation in a div.

  ### Component API

  | name     | description                                                    |
  | -------- | -------------------------------------------------------------- |
  | children | The content to display inside the component                    |
  | title    | The title to display when hovering the mouse over the ellipsis |
