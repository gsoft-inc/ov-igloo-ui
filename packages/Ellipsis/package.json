{
  "name": "@igloo-ui/ellipsis",
  "version": "0.0.1",
  "main": "dist/Ellipsis.js",
  "module": "dist/Ellipsis.js",
  "types": "dist/Ellipsis.d.ts",
  "styles": "dist/Ellipsis.css",
  "private": "true",
  "publishConfig": {
    "access": "public"
  },
  "repository": "git+https://github.com/gsoft-inc/ov-igloo-ui.git",
  "author": "Officevibe Inc.",
  "contributors": [
    "Sylvester sylvester.vuong@gsoft.com"
  ],
  "license": "Apache-2.0",
  "peerDependencies": {
    "react": ">= 16.8.6"
  },
  "files": [
    "dist"
  ],
  "dependencies": {
    "@igloo-ui/tokens": "^0.1.1",
    "classnames": "^2.3.1"
  }
}
