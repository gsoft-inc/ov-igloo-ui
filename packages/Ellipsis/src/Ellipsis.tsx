import * as React from 'react';
import { useRef } from 'react';
import classNames from 'classnames';

import './ellipsis.css';

export interface EllipsisProps {
  className?: string;
  containerClassName?: string;
  children?: React.ReactNode;
  title?: string;
  withTooltip?: boolean;
}

const Ellipsis: React.FunctionComponent<EllipsisProps> = (
  props: EllipsisProps
) => {
  const { children } = props;

  const parentElement = useRef<HTMLDivElement>(null);
  const element = useRef<HTMLDivElement>(null);

  if (!children) {
    return null;
  }

  return (
    <div
      className={classNames('ids-ellipsis', props.containerClassName)}
      title={props.title}
      ref={parentElement}
    >
      <div
        className={classNames('ids-ellipsis__content', props.className)}
        ref={element}
      >
        {children}
      </div>
    </div>
  );
};

export default Ellipsis;
