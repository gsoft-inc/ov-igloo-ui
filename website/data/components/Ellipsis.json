[
  [
    {
      "tags": {},
      "filePath": "/Users/franck.gaudin/Devel/Igloo/components/packages/Ellipsis/src/Ellipsis.tsx",
      "description": "",
      "displayName": "Ellipsis",
      "methods": [],
      "props": {
        "children": {
          "defaultValue": null,
          "description": "The content to display inside the ellipsis",
          "name": "children",
          "parent": {
            "fileName": "components/packages/Ellipsis/src/Ellipsis.tsx",
            "name": "EllipsisProps"
          },
          "declarations": [
            {
              "fileName": "components/packages/Ellipsis/src/Ellipsis.tsx",
              "name": "EllipsisProps"
            },
            {
              "fileName": "components/node_modules/@types/react/index.d.ts",
              "name": "TypeLiteral"
            }
          ],
          "required": false,
          "type": { "name": "ReactNode" }
        },
        "className": {
          "defaultValue": null,
          "description": "Add a className for the ellipsis content div",
          "name": "className",
          "parent": {
            "fileName": "components/packages/Ellipsis/src/Ellipsis.tsx",
            "name": "EllipsisProps"
          },
          "declarations": [
            {
              "fileName": "components/packages/Ellipsis/src/Ellipsis.tsx",
              "name": "EllipsisProps"
            }
          ],
          "required": false,
          "type": { "name": "string" }
        },
        "containerClassName": {
          "defaultValue": null,
          "description": "Add a className for the ellipsis container div",
          "name": "containerClassName",
          "parent": {
            "fileName": "components/packages/Ellipsis/src/Ellipsis.tsx",
            "name": "EllipsisProps"
          },
          "declarations": [
            {
              "fileName": "components/packages/Ellipsis/src/Ellipsis.tsx",
              "name": "EllipsisProps"
            }
          ],
          "required": false,
          "type": { "name": "string" }
        },
        "title": {
          "defaultValue": null,
          "description": "Add a title to the ellipsis",
          "name": "title",
          "parent": {
            "fileName": "components/packages/Ellipsis/src/Ellipsis.tsx",
            "name": "EllipsisProps"
          },
          "declarations": [
            {
              "fileName": "components/packages/Ellipsis/src/Ellipsis.tsx",
              "name": "EllipsisProps"
            }
          ],
          "required": false,
          "type": { "name": "string" }
        }
      }
    }
  ]
]
