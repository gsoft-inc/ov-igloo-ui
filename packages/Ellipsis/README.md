# @igloo-ui/ellipsis

Welcome to my `Ellipsis` component.

## Installation

```sh
npm i @igloo-ui/ellipsis
# or with yarn
yarn add @igloo-ui/ellipsis
```

Then to use the component in your code just import it!

```js
import Ellipsis from '@igloo-ui/ellipsis';
// and with css-modules
import '@igloo-ui/ellipsis/dist/ellipsis.css';
```
