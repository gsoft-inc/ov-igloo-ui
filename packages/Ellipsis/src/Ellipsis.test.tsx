/**
 * @jest-environment jsdom
 */
import React from 'react';
import { shallow } from 'enzyme';

import Ellipsis from './Ellipsis';

describe('Ellipsis', () => {
  const component = shallow(<Ellipsis>Hello world</Ellipsis>);
  test('It should render without errors', () => {
    const wrapper = component.find('.ids-ellipsis');
    expect(wrapper.length).toBe(1);
  });

  test('It should render a snapshot', () => {
    expect(component).toMatchSnapshot();
  });
});
