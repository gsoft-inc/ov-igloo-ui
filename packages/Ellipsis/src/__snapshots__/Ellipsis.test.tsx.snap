// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ellipsis It should render a snapshot 1`] = `ShallowWrapper {}`;
